src/artifact-uploader.cpp

```diff
@@ -19,28 +27,36 @@ namespace connect
 class ArtifactUploader::Impl
 {
 public:
+    Impl()
+        : accountId_(-1)
+        , cameraId_(-1)
+        , done_(false)
+        , timeoutToCompleteUploadSec_(0)
+    {
+    }
+
     ~Impl();
 
     Status init(const ArtifactUploader::Configuration& cfg,
                 ArtifactUploader::ClientConfigCallback* configCallback);
 
-    // all upload* methods are asynchronous, non-blocking, take ownership
-    // of data passed to them
-    Status uploadBackground(const timestamp_t& timestamp, PayloadHolderPtr payload);
-    Status uploadObjectStream(const ObjectStream& stream, PayloadHolderPtr payload);
-    Status uploadFlipbook(const Flipbook& flipbook, PayloadHolderPtr payload);
-    Status uploadEvent(const timestamp_t& timestamp, move_ref<Events> data);
-    Status uploadCount(move_ref<Counts> counts);
-
 private:
-    OutputControllerPtr outputController_;
-    UploadTaskQueuerPtr uploadTaskQueuer_;
+    friend class ArtifactUploader;
```

Comment:
How about just making an Impl::queue_push_back method instead of making a "friend".